### PR TITLE
07-multi-hop: port automated tests to pytest 

### PR DESCRIPTION
**Pull request by miri64**
This ports the existing automated tests for spec 7 to the new, `pytest`-based framework.

Requires #155

### CODE REVIEW

**Comment by miri64** [comment]
Rebased to current #155 (for which I removed the spec tests)

**Comment by miri64** [comment]
Rebased to current master.

**Comment by miri64** [comment]
Rebased to current master.

**Comment by miri64** [comment]
Rebased to current master.

File: 07-multi-hop/test_spec07.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        nodes_netifs.append({
+            "netif": key,
+            "hwaddr": netifs[key]["long_hwaddr"],
+            "lladdr": [addr["addr"] for addr in netifs[key]["ipv6_addrs"] if
+                       addr["scope"] == "link"][0],
+        })
+    return nodes_netifs
+
+
+# pylint: disable=W0621
```
Comment (fjmolinas) [inline 07-multi-hop/test_spec07.py:67]: Not needed.
```suggestion
```

File: 07-multi-hop/test_spec07.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            # set to-route
+            node.ifconfig_l2filter_add(nodes_netifs[i]["netif"],
+                                       nodes_netifs[i + 1]["hwaddr"])
+        if i > 0:
+            # set from-route
+            node.ifconfig_l2filter_add(nodes_netifs[i]["netif"],
+                                       nodes_netifs[i - 1]["hwaddr"])
+
+
+# pylint: disable=W0621
```
Comment (fjmolinas) [inline 07-multi-hop/test_spec07.py:80]:
```suggestion
```

**Comment by miri64** [comment]
>  can we open an issue for this instead? (same for `l2filter` stuff).

Huh? `l2filter` is in master already.

File: 07-multi-hop/test_spec07.py (hunk 2 above)
Comment (miri64) [reply 07-multi-hop/test_spec07.py:80]: Reasoning?

**Comment by miri64** [comment]
>  there is a todo mention, can we open an issue for this instead?

Why not both? I think it's good to keep the reminder in the code as well.

File: 07-multi-hop/test_spec07.py (hunk 2 above)
Comment (fjmolinas) [reply 07-multi-hop/test_spec07.py:80]: Pylint didn't complain when ran locally? Did I do it wrong?

**Comment by fjmolinas** [comment]
> Why not both? I think it's good to keep the reminder in the code as well.

Wrong choice of words, both is perfect.

**Comment by miri64** [comment]
> […] can we open an issue for this […]?

See https://github.com/RIOT-OS/RIOT/issues/14637

File: 07-multi-hop/test_spec07.py (hunk 2 above)
Comment (miri64) [reply 07-multi-hop/test_spec07.py:80]: Ah, you are right! `nodes_netifs` is used in those functions after all!

**Comment by miri64** [comment]
> Also seems like `_get_global_addr` and `_get_local_addr` could also be put in common somewhere.

You mean `_get_root_addr` with the second? There is no `_get_local_addr`.

**Comment by miri64** [comment]
> You mean `_get_root_addr` with the second? There is no `_get_local_addr`.

In that case I at least disagree for that one as this is very RPL specific and the only RPL tests are in this spec.

**Comment by miri64** [comment]
For global address I agree.

**Comment by fjmolinas** [comment]
> You mean _get_root_addr with the second? There is no _get_local_addr.

Sorry I think at the same time I was looking at the `native` `get_link_local`, and I meant it in the sense that both could exist for non-native.

**Comment by miri64** [comment]
> Sorry I think at the same time I was looking at the `native` `get_link_local`, and I meant it in the sense that both could exist for non-native.

There are no `native` tests here???!? But regarding `global_addr` there is a change in the making, I just run the tests for it now. Let's see how you like it :-).

**Comment by fjmolinas** [comment]
>   There are no native tests here???!? But regarding global_addr there is a change in the making, I just run the tests for it now. Let's see how you like it :-).

Yes I know I was looking at the `testutils` directory, there there are some native functions, and I saw `get_link_local`. Sorry for not being clear about that.

**Comment by miri64** [comment]
Have a look at the new integration of `global_addr` to `testutils`.

**Comment by fjmolinas** [comment]
Looks good please squash and rebase @miri64!

**Comment by miri64** [comment]
Done

**Pull request merged**